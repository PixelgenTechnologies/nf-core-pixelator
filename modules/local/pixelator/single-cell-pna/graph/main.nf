process PIXELATOR_PNA_GRAPH {
    tag "$meta.id"
    label 'process_high_memory'
    label 'process_long'

    // TODO: Add conda
    // conda "bioconda::pixelator=0.18.2"

    // TODO: Add containers
    // container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
    //     'https://depot.galaxyproject.org/singularity/pixelator:0.18.2--pyhdfd78af_0' :
    //     'biocontainers/pixelator:0.18.2--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(edge_list), path(panel_file), val(panel)

    output:
    tuple val(meta), path("graph/*")                    , emit: all_results
    tuple val(meta), path("graph/*.pxl")                , emit: pixelfile
    tuple val(meta), path("graph/*.report.json")        , emit: report_json
    tuple val(meta), path("graph/*.meta.json")          , emit: metadata_json
    tuple val(meta), path("*pixelator-graph.log")       , emit: log

    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:

    prefix = task.ext.prefix ?: "${meta.id}"
    def args = task.ext.args ?: ''
    def panelOpt = (
        panel ? "--panel $panel" :
        panel_file ? "--panel $panel_file" : ""
    )

    """
    pixelator \\
        --cores $task.cpus \\
        --log-file ${prefix}.pixelator-graph.log \\
        --verbose \\
        single-cell-pna \\
        graph \\
        $panelOpt \\
        --output . \\
        $args \\
        ${edge_list}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        pixelator: \$(echo \$(pixelator --version 2>/dev/null) | sed 's/pixelator, version //g' )
    END_VERSIONS
    """
}
