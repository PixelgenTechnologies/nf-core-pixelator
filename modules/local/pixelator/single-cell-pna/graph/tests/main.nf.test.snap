{
    "versions": {
        "content": [
            [
                "versions.yml:md5,014a0d8870a21fc70b6025ee7813a4cd"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-06T15:25:29.63045379"
    },
    "report_json": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.report.json:md5,44c0c42d4dbb382f7d1541719182ca08"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-16T15:56:26.49718139"
    },
    "metadata_json": {
        "content": [
            [
                [
                    {
                        "id": "PNA055_Sample07_filtered_S7",
                        "design": "pna-2",
                        "panel": "proxiome-immuno-155",
                        "technology": "pna"
                    },
                    "PNA055_Sample07_filtered_S7.meta.json:md5,b15fffdd71ae59f5e36f7237e76b1b98"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-16T15:56:26.489197437"
    }
}